retry/src/main/java/com/iluwatar/retry/Retry.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * The MIT License (MIT)
+ * 
+ * Copyright (c) 2014-2016 Ilkka Seppälä
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.iluwatar.retry;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Predicate;
+
+/**
+ * Decorates {@link BusinessOperation business operation} with "retry" capabilities.
+ *
+ * @author George Aristy (george.aristy@gmail.com)
+ * @param <T> the remote op's return type
+ * @since 1.18.0
+ */
+public final class Retry<T> implements BusinessOperation<T> {
+  private final BusinessOperation<T> op;
+  private final int maxAttempts;
```

Comment:
by default make retry = 1?

Reply:
@zafarella `Retry` already executes the op at least once. Do you want it to execute it at least twice?

I like to default to just 1 execution since not all operations may be qualified for retries.

Reply:
👍